src/main/java/org/dcsa/core/exception/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,95 @@
+package org.dcsa.core.exception.handler;
+
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import lombok.extern.slf4j.Slf4j;
+import org.dcsa.core.exception.*;
+import org.springframework.core.codec.DecodingException;
+import org.springframework.http.HttpStatus;
+import org.springframework.r2dbc.BadSqlGrammarException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.server.ServerWebInputException;
+import javax.validation.ConstraintViolationException;
+
+@Slf4j
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+  @ExceptionHandler(DCSAException.class)
+  public void handleDCSAExceptions(DCSAException dcsaEx) {
+    log.error(
+        "{} ({}) - {}",
+        this.getClass().getSimpleName(),
+        dcsaEx.getClass().getSimpleName(),
+        dcsaEx.getMessage(),
+        dcsaEx);
+    throw dcsaEx;
+  }
```

Comment:
This probably wants to be trace logging (or debug logging)  If we get an error log with a full stack trace everytime someone submits an invalid request, then the logfile will grow insanely fast.

Reply:
Agreed. Added it into trace logs, if enabled.